Shah docs 11

ShahFragcolor opened the pull request:
- Documented Math shards so that they have specific help text
- Replaced returns with outputs


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
Documented Math shards

- Created structs for shards in Math.cpp so that they have specific help text instead of using the standard one.

ShahFragcolor committed:
Replaced "returns" to "outputs"

ShahFragcolor committed:
Corrected Typo

ShahFragcolor committed:
Fixed Math.Negate

Accidentally broke it, was trying to use UnaryVarOperation instead of UnaryOperation

sinkingsugar commented on shards/modules/core/math.hpp:
```diff
@@ -536,38 +550,238 @@ template <class TOp> struct UnaryVarOperation final : public UnaryOperation<TOp>
-  };                                                                                     \
-  using NAME = UnaryOperation<BasicUnaryOperation<NAME##Op, DispatchType::NumberTypes>>; \
-  RUNTIME_SHARD_TYPE(Math, NAME);
+// MATH_BINARY_OPERATION(Add, +, 0);
```
this can be removed now no?

sinkingsugar commented on shards/modules/core/math.hpp:
```diff
@@ -602,6 +816,427 @@ MATH_UNARY_FLOAT_OPERATION(Floor, __builtin_floor, __builtin_floorf);
+struct Acos : public UnaryFloatOperation<BasicUnaryOperation<AcosOp, DispatchType::FloatTypes>> {
+  static SHOptionalString help() {
+    return SHCCSTR(
+        "This shard calculates the inverse cosine of the given input (acos(x)), where the input x is the cosine value.");
```
Foreign math notation not needed

sinkingsugar commented on shards/modules/core/math.hpp:
```diff
@@ -602,6 +816,427 @@ MATH_UNARY_FLOAT_OPERATION(Floor, __builtin_floor, __builtin_floorf);
+struct Sinh : public UnaryFloatOperation<BasicUnaryOperation<SinhOp, DispatchType::FloatTypes>> {
+  static SHOptionalString help() {
+    return SHCCSTR("This shard calculates the hyperbolic sine of the given input (sinh(x)), where the input x is the real number "
+                   "used in the exponential calculation (sinh(x) = (e^x - e^-x) / 2).");
```
Too much Foreign math notation all over the place, let's use Shards notation or replace with words

sinkingsugar commented:
Almost good to go

ShahFragcolor committed:
Made requested change to remove math special symbols

- Removed math equations and special symbols from all help text and replaced them with words instead.

sinkingsugar committed:
Update math.hpp
